src/java/picard/analysis/GcBiasMetricsCollector.java

```diff
@@ -191,7 +191,7 @@ public void addMetricsToFile(final MetricsFile<GcBiasMetrics, Integer> file) {
                 final long[] errorsByGc = gcCur.errorsByGc;
                 final long[] basesByGc = gcCur.basesByGc;
                 final int totalClusters = gcCur.totalClusters;
-                final int totalAlignedReads = gcCur.totalAlignedReads;
+                final long totalAlignedReads = gcCur.totalAlignedReads;
```

Comment:
Yes you're right, that part fixes the bug. I'll just implement the int->long change.